### PR TITLE
Global Styles: Add Help Link to Plan Limitation Message

### LINKED ISSUE #93401: Add Help Link to Premium Styles Plan Limitation Message
### What

Add the below support link to the warning message displayed to users who edit Styles without an eligible plan:

> Your site includes premium styles that are only visible to visitors after upgrading to the Premium plan or higher.

https://wordpress.com/support/using-styles/#access-to-styles

### Why

The message alone is confusing to some users who don't understand why the styles are gated. We can alleviate this confusion with a link to more context on our /support site. Example: 8569729-zd-a8c

### How

Could use a "Learn more" option next to the text:
Screenshot 2024-08-09 at 12 40 40 PM

### PR DESCRIPTION
**Pull request by epeicher**
Resolves https://github.com/Automattic/wp-calypso/issues/93401

## Proposed changes:

* Adds a new link to the plan limitation notice message on the editor

||Before | After| Clicking the link|
|-------|-------|------|------|
|Site Editor| CleanShot 2024-09-05 at 13 35 39@2x |CleanShot 2024-09-06 at 11 41 28@2x |      CleanShot 2024-09-06 at 11 41 45@2x       |
|Post Editor| CleanShot 2024-09-23 at 18 33 16@2x |    CleanShot 2024-09-23 at 18 31 16@2x       |  CleanShot 2024-09-23 at 18 32 46@2x  |

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply these changes using [the instructions below](https://github.com/Automattic/jetpack/pull/39226#issuecomment-2328576358)
* Navigate to a site that has a lower than Premium plan, e.g. free
* Sandbox your environment. If you are using a Simple Site, sandbox the site
- Navigate to the Styles editor, make sure your site has a theme that allows it, for example Twenty Twenty Four theme. The URL should be `https://<your-site>/wp-admin/site-editor.php?path=%2Fwp_global_styles`
- Change the Style and click on the main editor to open it
- Check the top notice is displayed, and it contains a new link `Learn more`. Click on the link and check the Help Center opens with the URL  https://wordpress.com/support/using-styles/#access-to-styles
- Save your changes
* Navigate to any Post or create a new one
* Check the top notice is displayed, and it contains a new link `Learn more`. Click on the link and check the Help Center opens with the URL  https://wordpress.com/support/using-styles/#access-to-styles

### CODE REVIEW

**Review by miksansegundo — DISMISSED** [review]
LGTM 👍 

I'm not sure about using the question mark icon (?) in the "Learn more" link because we don't seem to use it.

See for reference:

Screenshot 2567-09-06 at 12 54 22

Screenshot 2567-09-06 at 12 48 47

Screenshot 2567-09-06 at 12 54 32

### Follow-up improvement

Opening the help center within the editor would be a nice improvement. See an example [here](https://github.com/Automattic/jetpack/blob/03f859c123bc185768a8619c08f637c97b25467f/projects/packages/jetpack-mu-wpcom/src/features/wpcom-block-description-links/index.tsx#L20) that is using the component [DescriptionSupportLink](https://github.com/Automattic/jetpack/blob/03f859c123bc185768a8619c08f637c97b25467f/projects/packages/jetpack-mu-wpcom/src/features/wpcom-block-description-links/src/inline-support-link.tsx#L26C25-L26C47).

For another reference, see the component [InlineSupportLink](https://github.com/Automattic/wp-calypso/tree/a30fcc6021c9e68a69477ee73ee8c2af227369c8/client/components/inline-support-link) but that lives in Calypso.

**Comment by epeicher** [comment]
Thanks, @miksansegundo, for the recommendations. I have followed your suggestions, and I have been able to display the URL of the link on the Help Center as part of these changes: https://github.com/Automattic/jetpack/pull/39226/commits/9c412adf6a1e4a900caf62fb52c8ddc0c640ba7e. The PR is ready for another review 🙌 

@mmtr, @taipeicoder, it would be great if you could review the changes as you have been involved in some previous changes related to this functionality 🙏

**Review by mmtr — COMMENTED** [review]
Works great! But I think we need to revisit how the warning looks in the post editor. We display different actions there, and the addition of the new "Learn more" action looks a bit off:

Screenshot 2024-09-12 at 17 56 28

I think we can probably remove the "Remove premium styles" action, since it links to a section that users can already find via the new "Learn more" action.

**Comment by epeicher** [comment]
Thanks for your suggestion, @mmtr! I agree with that. It looks off, and both actions, `Remove premium styles` and `Learn more` open a similar link, so I will go ahead and remove the `Remove premium styles` link from the post editor.

**Comment by epeicher** [comment]
Hi @mmtr, I have applied your suggestions that apply to the Post editor as part of [0a4afe3](https://github.com/Automattic/jetpack/pull/39226/commits/0a4afe33545449745e25705f5ff8faf925e01ad9) and the changes are ready for another review. 

As you can see, I have also removed the leftovers after the changes. I have checked that they are not used, but please let me know if you find any regressions or other potential changes that could be done. Thanks!

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/notices.js [hunk 1]
```diff
@@ -175,14 +187,20 @@ function GlobalStylesEditNotice() {
...
-			label: __( 'Remove premium styles', 'jetpack-mu-wpcom' ),
-			onClick: isSiteEditor ? resetGlobalStyles : openResetGlobalStylesSupport,
-			variant: isSiteEditor ? 'secondary' : 'link',
+			label: __( 'Learn more', 'jetpack-mu-wpcom' ),
+			onClick: openLearnMoreAboutStylesDialog,
+			variant: 'link',
 			noDefaultClasses: true,
-			className: isSiteEditor
-				? ''
-				: 'wpcom-global-styles-action-has-icon wpcom-global-styles-action-is-external wpcom-global-styles-action-is-support',
```
Comment (mmtr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/notices.js:185]: The `wpcom-global-styles-action-is-support` class is no longer used, so we can clean up some unused CSS code now:

https://github.com/Automattic/jetpack/blob/671d537286be54d41935a1db6991e8645aa74a77/projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/notice.scss#L80-L87

**Review by mmtr — DISMISSED** [review]
Works great! Just left a minor note about removing some unused code.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/notices.js (hunk 1 above)
Comment (epeicher) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/notices.js:185]: Thanks for pointing that out, I have removed that unused style in this commit [4e640eb](https://github.com/Automattic/jetpack/pull/39226/commits/4e640eb2777c05860a752de6bcd97ab20c82fd2d)

**Review by mmtr — APPROVED** [review]

**Comment by epeicher** [comment]
I have done a quick testing in atomic sites after the release p1727207048460609-slack-C05GLGHLM8U, and I have not found any issues 🙌

**Pull request merged**